tests/unit/fem/test_assemblediagonalpa.cpp

```diff
@@ -357,9 +355,10 @@ TEST_CASE("Hcurl/Hdiv diagonal PA",
 
          enum Spaces {Hcurl, Hdiv};
 
-         for (int spaceType = 0; spaceType < numSpaces; ++spaceType)
+         for (int spaceType = 0; spaceType < 2; ++spaceType)
          {
-            const int numIntegrators = (dimension == 3 || coeffType < 2) ? 2 : 1;
+            const int numIntegrators = ((dimension == 3 || coeffType < 2) &&
+                                        spaceType == 0) ? 2 : 1;
```

Comment:
This logic is a little complicated. Can you add a comment explaining?

Reply:
One reason it was complicated is that the case of the Hdiv space with coeffType 1 (non-constant scalar coefficient) was not working, since the `DivDivIntegrator` constructor did not take an `IntegrationRule` as an argument, and PA and full assembly were using different rules. I fixed this so that test passes. Now the only tests being skipped are for non-scalar coefficients for div-div or 2D curl-curl, and that is documented.

Reply:
That makes sense, thanks. It's clearer now 👍